homeassistant/components/mqtt/discovery.py

```diff
@@ -321,6 +254,81 @@ async def async_integration_message_received(integration, msg):
             )
 
 
+async def async_process_discovery_payload(
+    hass, config_entry, component, discovery_id, payload
+):
+    """Process the payload of a new discovery."""
+
+    _LOGGER.debug("Process discovery payload %s", payload)
+    discovery_hash = (component, discovery_id)
+    if discovery_hash in hass.data[ALREADY_DISCOVERED] or payload:
+
+        async def discovery_done(_):
+            pending = hass.data[PENDING_DISCOVERED][discovery_hash]["pending"]
+            _LOGGER.debug("Pending discovery for %s: %s", discovery_hash, pending)
+            if not pending:
+                hass.data[PENDING_DISCOVERED][discovery_hash]["unsub"]()
+                hass.data[PENDING_DISCOVERED].pop(discovery_hash)
+            else:
+                payload = pending.pop()
+                await async_process_discovery_payload(
+                    hass, config_entry, component, discovery_id, payload
+                )
+
+        if discovery_hash not in hass.data[PENDING_DISCOVERED]:
+            hass.data[PENDING_DISCOVERED][discovery_hash] = {
+                "unsub": async_dispatcher_connect(
+                    hass,
+                    MQTT_DISCOVERY_DONE.format(discovery_hash),
+                    discovery_done,
+                ),
+                "pending": deque([]),
+            }
+
+    if discovery_hash in hass.data[ALREADY_DISCOVERED]:
+        # Dispatch update
+        _LOGGER.info(
+            "Component has already been discovered: %s %s, sending update",
+            component,
+            discovery_id,
+        )
+        async_dispatcher_send(
+            hass, MQTT_DISCOVERY_UPDATED.format(discovery_hash), payload
+        )
+    elif payload:
+        # Add component
+        _LOGGER.info("Found new component: %s %s", component, discovery_id)
+        hass.data[ALREADY_DISCOVERED][discovery_hash] = None
+
+        config_entries_key = f"{component}.mqtt"
+        async with hass.data[DATA_CONFIG_ENTRY_LOCK]:
+            if config_entries_key not in hass.data[CONFIG_ENTRY_IS_SETUP]:
+                if component == "device_automation":
+                    # Local import to avoid circular dependencies
+                    # pylint: disable-next=import-outside-toplevel
+                    from . import device_automation
+
+                    await device_automation.async_setup_entry(hass, config_entry)
+                elif component in "tag":
```

Comment:
Why is this changed to `in` instead of `==`?

Reply:
Don't known, corrected that.